### PR TITLE
Add post forms

### PR DESCRIPTION
**Pull request by eng-mohamed-eldeeb**
## in this project I did

- Create a method called current_user in the ApplicationController to make current user data available to all controllers.
It returns the first user from the database.

- Create forms to perform the following functions:
>  -Creates a Post on behalf of the current_user (use the method that I created in the ApplicationController).
> - Create a comment on behalf of the current_user (use the method that I created in the ApplicationController).

### CODE REVIEW

File: app/views/comments/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+          <% end %>
+        </div>
+      <% end %>
+      <div class="mb-4 flex flex-col">
+        <%= form.text_field :text, placeholder: "Add new comment here", class: "text-[#11537d] p-3" %>
+      </div>
+      <div class="mb-4 flex flex-col">
+        <%= form.submit "Comment" , class: "text-[#11537d] text-xl font-medium bg-white px-4 py-2 rounded-[15px] w-fit cursor-pointer" %>
+      </div>
+<% end %>
```
Comment (Deepakdanger) [inline app/views/comments/new.html.erb:16]: - [ ] I can't able to see the form. Please fix that.

File: app/views/posts/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  </div>
+  <div class="mb-4 flex flex-col">
+    <%= form.label :text %>
+    <%= form.text_field :text, placeholder: "Add post body", class: "p-2" %>
+  </div>
+  <div class="mb-4 flex flex-col">
+    <%= form.submit "Add post" , class: "text-white text-xl font-medium bg-[#11537d] px-6 py-3 rounded-[15px] w-fit cursor-pointer" %>
+  </div>
+<%end%>
+</div>
```
Comment (Deepakdanger) [inline app/views/posts/new.html.erb:22]: - [ ] I can't able to see the posts form. Please fix that.

File: app/views/posts/show.html.erb [hunk 3]
```diff
@@ -3,7 +3,16 @@
...
+
+
+<%= form_with model: [@post, @like], url: create_post_like_path  do |form| %>
+      <div class="mb-4 flex flex-col cursor-pointer">
+      <div class="text-[#11537d] text-xl font-medium flex gap-2 bg-white px-4 py-2 rounded-[15px] w-fit cursor-pointer">
+<p class="self-end"><%= @post.likes_counter %></p>
+        <%= form.submit "Like 👍" %>
+      </div>
+      </div>
+    <% end %>
```
Comment (Deepakdanger) [inline app/views/posts/show.html.erb:15]: - [ ] I can't able to like the post. Please fix that.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @eng-mohamed-eldeeb, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 

## Required Changes ♻️
- [ ] In the user index form, the title of the post is not readable. You can see in the below pic. It's above the "**See all posts**". Please fix this.

Screenshot from 2023-01-28 22-26-35

- [ ] I am unable to create anything as any of the buttons or forms are not visible in the UI. Please fix.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by fobadara — APPROVED** [review]
Hi @eng-mohamed-eldeeb, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ Linter are passing
✅ Implemented reviewer suggestions

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**